src/Compilers/VisualBasic/Test/Semantic/Semantics/Conversions.vb

```diff
@@ -5095,5 +5097,42 @@ End Module
 ]]>)
         End Sub
 
+        <WorkItem(73032, "https://github.com/dotnet/roslyn/issues/73032")>
+        <Fact()>
+        Public Sub ConvertLargeDoubleConstantsAndLiteralsToLong()
+            Dim compilation = CreateCompilationWithMscorlib40AndVBRuntime(
+    <compilation>
+        <file name="a.vb"><![CDATA[
+Module Program
+    Sub Main()
+        System.Console.WriteLine(CType(CDbl(&H8000000000000000L), Long))
```

Comment:
>System.Console.WriteLine(CType(CDbl(&H8000000000000000L), Long)) [](http://example.com/codeflow?start=8&length=64)

Instead of outputting the values, I think we should compare them with result of conversions performed at runtime. For example it is not obvious to me that the values in the output are correct. I think the overflow check for VB is set to true by default, but it would be good to set this option explicitly (`WithOverflowChecks`).

Reply:
In each case, the Long argument to CDbl is the expected result of calling CType. So I could write
```
System.Console.WriteLine(CType(CDbl(&H8000000000000000L), Long) = &H8000000000000000L)
```
and verify that the output is True. Is that acceptable?

Reply:
This is going to rely on compile time constant folding for equality, which would be desirable to avoid. The goal is to compare result of compile time constant folding with result that doesn't rely on constant folding. I have some thing like the following in mind:
```
 System.Console.WriteLine(CType(CDbl(&H8000000000000000L), Long) = ConvertToLong(CDbl(&H8000000000000000L)))


Function ConvertToLong(x as Double) As Long
    Return CType(x, Long)
End Function
```